app/classes/query/modules/sequence_operators.rb

```diff
@@ -60,18 +59,23 @@ def prev
         new_self.current_id = result_ids[index - 1]
       end
     elsif has_outer?
-      while new_self = new_self.outer_prev
-        if new_new_self = new_self.last(:skip_outer)
-          new_self = new_new_self
-          break
-        end
-      end
+      new_self = prev_inner(new_self)
```

Comment:
How is this related to updating the gems?

Reply:
See below.